data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSource.java

```diff
@@ -102,7 +104,10 @@ public void start(Buffer<Record<Object>> buffer) {
 
             final ServerBuilder sb = Server.builder();
             sb.disableServerHeader();
-            sb.service(grpcServiceBuilder.build());
+
+            final Function<? super HttpService, ? extends HttpService> httpAuthenticationService = authenticationProvider.getHttpAuthenticationService().get();
```

Comment:
This will throw an exception when the `Optional` is empty. This code should be refactored to check that the `Optional` is present. And only if it is add it to the service builder.